src/and-cli-restore-azure-storage.ts

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env node
+
+import { CommandRunner } from "./modules/command-runner";
+import { Echo } from "./modules/echo";
+import program from "commander";
+import shell from "shelljs";
+import { AzureAzcopySync } from "./modules/azure-azcopy-sync";
+
+CommandRunner.run(async () => {
+    // -----------------------------------------------------------------------------------------
+    // #region Variables
+    // -----------------------------------------------------------------------------------------
+
+    let deleteDestination: boolean = false;
+    let destinationAccount: string;
+    let destinationContainer: string;
+    let destinationSasToken: string;
+    let recursive: boolean = false;
+    let sourceAccount: string;
+    let sourceContainer: string;
+    let sourceSasToken: string;
+
+    // #endregion Variables
+
+    /// -----------------------------------------------------------------------------------------
+    // #region Functions
+    // -----------------------------------------------------------------------------------------
+
+    const restoreAzureStorage = {
+        description() {
+            return "Restore application assets in Azure Storage";
+        },
+        run() {
+            // Check system/command requirements
+            this.validateOrExit();
+
+            /*
+                Following options currently only support syncing utilizing Azure SAS Tokens.
+                Consider future refactor to allow this restore aspect to first generate a SAS
+                Token using a different means of authentication (service provider, active
+                directory, etc...).  Consider using AZ login and a service provider to then
+                generate a SAS before passing that SAS Token into the Azcopy module later instead
+                of forcing the consumer to define a hard coded value or force the consumer to
+                generate one on its own cadence.
+            */
+            const options = {
+                deleteDestination: deleteDestination,
+                destination: {
+                    account: destinationAccount,
+                    container: destinationContainer,
+                    sasToken: destinationSasToken,
+                },
+                recursive: recursive,
+                source: {
+                    account: sourceAccount,
+                    container: sourceContainer,
+                    sasToken: sourceSasToken,
+                },
+            };
+
+            AzureAzcopySync.containers(options);
+
+            Echo.newLine();
+            Echo.success(
+                "Containers successfully synced in Azure Blob Storage!"
```

Comment:
Suggestion/Question: Is it worth adding a more detailed success message here? Something along the lines of "Contents of `options.source.container` successfully synced to `options.destination.container` in Azure Blob Storage!"? Just wondering if that would give any additional value to the consumer.